### ISSUE TITLE
reduce map tile consumption on the home page - 5

### ISSUE THREAD

**Issue by paregorios** [isawnyu/pleiades-gazetteer#361]
Our mapbox costs are increasing because of number of tile hits. Can we modify the home page to reduce impact? We had the following suggestions from a mapbox person:

> I think there are some things you can do on your end to reduce the amount of map views your account sees each month. Map views are totally dependent on how many map tiles your end users load. Panning, scrolling, and zooming are all interactions that cause more map tiles to load. As an example, your maps at https://pleiades.stoa.org/ scroll automatically every few seconds to new locations. This scrolling action causes more map tiles to load which leads to more map views on your account. 
>
> Additionally, searching for key terms like "settlement" pulls up several Mapbox maps. Each of these maps add their own API calls and contribute to your overall map view count. 
> 
> ![image](https://user-images.githubusercontent.com/263285/54848296-f03c1280-4cae-11e9-8fa6-1c11fa6e393e.png)

Easy things to do in this regard might include:

 - Limit the number of example places that show up in the top-of-home-page map and then just loop over the same ones again and again.
 - Increase the delay on loading maps in the search results unless/until the user hits the "enter" key in the search box. We can go ahead and populate results dynamically as now, but delay a couple seconds trying to load the maps.
 - Batch the search results by some sane number so that if one gets a huge result, it doesn't try to load maps for all the results, just the ones in the current batch.

**Comment by skleinfeldt** [comment]
To cut down on the Mapbox cost of showing maps in response to a search, would it be useful/feasible to consider automatically creating a static image of the standard map view of an item, storing it with the item, and using that in search results?

**Comment by rmhorne** [comment]
@skleinfeldt - that is a really interesting idea. We can probably use "good enough" resolution for these images, and write some sort of script to auto-generate them when the representative point changes...

**Comment by paregorios** [comment]
This is potentially useful. I've often thought there are many things we could do with pre-composed map images.

I think we should prioritize as follows (guessing about which solution combos are the cheapest to implement):

 - Modify front-page map usage:
    - Increase time delays between each successive (random) item shown on the front page map.
    - Cap the number of places to be shown at 10-15 and once they are done, just repeat the zoom-and-pan for these few. 
     - Stop zoom-and-pan display entirely when a user clicks on an item in the search results and/or when the tab containing the front-page loses focus

Investigate a mechanism for on-update creation of raster map images for each place. Then look at using them in:
 - place and location pages
 - search results

**Comment by paregorios** [comment]
Story:

As a Pleiades user, when I first load the front page, I will see the map animation as now, however, the following changes in behavior will obtain:

 - After pan-zooming to 10 randomly selected places, that list will repeat until the map animation is stopped for some other reason.
 - Stop the map animation once I start typing in the search box. Do not restart the animation unless I reload the page.
 - Stop the map animation once I select any link on the page. Do not restart the animation unless I reload the page.

**Comment by MatthewWilkes** [comment]
I know this part isn't in scope for this story, but I wanted to add that the raster map option is unlikely to be possible due to the Mapbox terms of service:

    you may not redistribute any map content from the Services, including from a cache, by proxying, or by using a screenshot or other static image, instead of directly accessing via our APIs

**Comment by paregorios** [comment]
Currently, we pay mapbox for quantity of raster tile load (via the Raster Tiles API). The new pricing model works around "map loads" regarding which they say:

> A map load is counted every time Mapbox GL JS initializes on a webpage or in a web app. A map load includes unlimited Vector Tiles API and Raster Tiles API requests.

Per month, we'd get "up to 50,000" map loads for free. According to old Google Analytics data, we only run about 25,000 - 35,000 __page__ views per month. So, I think the first thing we ought to do is to convert to the new pricing, which involves a Javascript library change. I'll create a new ticket for that.

**Comment by paregorios** [comment]
Blocked by #405

**Comment by paregorios** [comment]
Obviated by change to map views.
